scalyr_agent/monitor_utils/k8s.py

```diff
@@ -17,6 +22,624 @@ class K8sApiException( Exception ):
     """
     pass
 
+class PodInfo( object ):
+    """
+        A collection class that stores label and other information about a kubernetes pod
+    """
+    def __init__( self, name='', namespace='', uid='', node_name='', labels={}, container_names=[], annotations={}, deployment_name=None ):
+
+        self.name = name
+        self.namespace = namespace
+        self.uid = uid
+        self.node_name = node_name
+        self.labels = labels
+        self.container_names = container_names
+        self.annotations = annotations
+        self.deployment_name = deployment_name
+
+        # generate a hash we can use to compare whether or not
+        # any of the pod info has changed
+        md5 = hashlib.md5()
+        md5.update( name )
+        md5.update( namespace )
+        md5.update( uid )
+        md5.update( node_name )
+
+        # flatten the labels dict in to a single string
+        flattened = []
+        for k,v in labels.iteritems():
+            flattened.append( k )
+            flattened.append( v )
+        md5.update( ''.join( flattened ) )
```

Comment:
Why not just move the `md5.update` up into the loop?  It seems a little unnecessary to create flattened.

Reply:
Update requires a string.  If you just call str( labels ) then if labels is a JsonObject you get the object id and type rather than a key/value textual representation of the object.

Flattening them out like this ensures the md5 is based on the contents of the key/value strings.

There is no guarantee of ordering in a dict, but for the same set of keys it seems to be consistent, at least within a single running instance.